File: dotnet/xmsclient/Client/State/IllegalStateTransitionException.cs
using System;
using jpmorgan.mina.common;
using log4net;

namespace OpenAMQ.XMS.Client.State
{
    public class IllegalStateTransitionException : AMQException
    {
        private AMQState _originalState;

        private Type _frame;

        public IllegalStateTransitionException(AMQState originalState, Type frame)
            : base("No valid state transition defined for receiving frame " + frame +
                   " from state " + originalState)
        {            
            _originalState = originalState;
            _frame = frame;
        }

        public AMQState OriginalState            
        {
            get
            {
                return _originalState;
            }
        }

        public Type FrameType
        {
            get
            {
                return _frame;
            }
        }
    }
}

File: java/src/org/openamq/client/state/IllegalStateTransitionException.java
package org.openamq.client.state;

import org.openamq.AMQException;

/**
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class IllegalStateTransitionException extends AMQException
{
    private AMQState _originalState;

    private Class _frame;

    public IllegalStateTransitionException(AMQState originalState, Class frame)
    {
        super("No valid state transition defined for receiving frame " + frame +
              " from state " + originalState);
        _originalState = originalState;
        _frame = frame;
    }

    public AMQState getOriginalState()
    {
        return _originalState;
    }

    public Class getFrameClass()
    {
        return _frame;
    }
}
